extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/VertxRequestHandler.java

```diff
@@ -114,6 +136,19 @@ private void dispatch(RoutingContext routingContext, InputStream is, VertxOutput
             if (!vertxRequest.getAsyncContext().isSuspended()) {
                 try {
                     vertxResponse.finish();
+                    // the reason we only cache successful responses is to ensure that a torrent of erroneous URLs
+                    // doesn't fill up the cache and cause a DoS
+                    if (setInitDataUponSuccess && vertxResponse.getStatus() >= 200 && vertxResponse.getStatus() <= 300) {
+                        // we don't want the cache to grow unbounded since values path params could potentially be infinite
+                        // and if left unchecked would take up all memory if the application is up for long enough
+                        if (resteasyUriInfoInitDataMap.size() > resteasyUriInfoCacheMaxSize) {
+                            resteasyUriInfoInitDataMap.clear(); // this is super lame and should probably be revisited
```

Comment:
yes it's dangerous. See my other comments on the design discussion.

Reply:
If we have the RESTEasy part of the PR in which puts in the necessary hooks, we can absolutely improve this based on what you mention about the existence or not of Caffeine.
Note that the way I have done the RESTEasy PR (and as of course you have seen from looking at this PR), the caching itself is handled exclusively in Quarkus so we can probably get as creative as we like.